### PR TITLE
chore: migrate shared-web to subtree

### PR DESCRIPTION
**Pull request by trentwatt**
## Description

convert existing submodule at `webui/react/src/shared` to be a subtree in order for us to try it out a bit.

incorporating `shared-web` git history into `core` resulted in some hassles when switching branches, so left that out. history in actual `shared-web` repo  remains intact.

## Test Plan

`git fetch upstream`
`git checkout subtree`

switch back and forth between any branches you have locally (especially older ones) and `subtree`, check that status is clean and you don't get blocked. there may be some issues that are config and/or local-residue dependent. check if issues persist with fresh clone of repo.

### test rebasing

`git remote add hamid https://github.com/hamidzr/determined`
`git checkout share-eslint`
`git rebase subtree`
`git rebase --skip` (for the commits that only touch `shared`)
`git subtree pull --prefix webui/react/src/shared https://github.com/determined-ai/shared-web share-eslint-styles --squash`

[here is a PR reflecting the result of that process](https://github.com/hamidzr/determined/pull/11/files). there are some artifacts outside of web, but looks like that's just because existing branch is not up to date.

### test PRs

[example PR that encompasses core and shared](https://github.com/determined-ai/determined/pull/4403). Can merge it, and then rebase `share-eslint` again.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hamidzr — COMMENTED** [review]
lgtm thanks for making the change. I think we should try to fix shared/bin/test.py before we land so that
1. we can partially test the changes against saas when we're making changes in core (ci won't do this here easily because  secret sharing complexity)
2. saas ci jobs won't start failing when they test that core builds fine.

**Review by hamidzr — APPROVED** [review]
looks good to me a couple of notes.
- should we add some details to the read me to tell people about the subtree, the commands they might want or the process? as in they should pull in changes and push those back up after to the shared web.
- might make sense to squash all the old shared-web commits into one so it's easier to follow the pr history if we're squash merging into core anyway

File: webui/react/src/shared/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Limitations:
+- shared dependencies between all the projects would need to have similar APIs
+- no pre-canned component-level versioning solution
+
+Read more about the methodology and tradeoffs [here](https://hpe-my.sharepoint.com/:w:/p/hamid_zare/EZh809x5395CrmypPcu8yiUBO4_Pk1cK_MdRnP3bywOLrQ?e=VdccG4)
+
+## How
+
+Code in this directory should not depend on internal dependencies that leave outside this directory
```
Comment (dannysauer) [inline webui/react/src/shared/README.md:21]:
```suggestion
Code in this directory should not depend on internal dependencies that live outside this directory
```
Typo?

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+shared-web:
+- webui/react/src/shared/**/*
```
Comment (dannysauer) [inline .github/labeler.yml:2]:
```suggestion
- webui/react/src/shared/**
```
You only need the trailing `/*` if you want something like `*.js` in any directory. If it's "all the files in any subdirs", then /** matches.

I'm pretty sure it'll still work the way it is, but if the Readme spelling is fixed, may as well clean this up at the same time. :)

**Pull request merged**